docs/reference/developer/notebook-style-guide.ipynb

```diff
@@ -0,0 +1,78 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "3205d1c4-8704-4b77-80b6-287e252864eb",
+   "metadata": {},
+   "source": [
+    "# Jupyter Notebook Style Guide\n",
+    "\n",
+    "## Motivation\n",
+    "\n",
+    "This style guide is meant for anyone writing documentation pages, tutorials, or workflows as Jupyter notebooks.\n",
+    "\n",
+    "The purpose is to make notebooks easy to understand and readable, and to minimize the reader's cognitive overload when switching between reading various notebooks.\n",
+    "\n",
+    "## Recommendations\n",
+    "\n",
+    "### Saving notebooks\n",
+    "\n",
+    "- Avoid storing outputs before saving or committing a notebook.\n",
+    "  - Use \"Restart kernel and clear all outputs\" before saving.\n",
+    "  - [nb-clean](https://github.com/srstevenson/nb-clean) can be used to do this automatically.\n",
+    "\n",
+    "### Document structure\n",
+    "\n",
+    "- Use markdown sections, subsections, ... for easy navigation and readability.\n",
+    "    \n",
+    "### Portability\n",
+    "\n",
+    "- Avoid hard-coding file paths which would require users to download files into specific folders.\n",
+    "- Use APIs, not file systems, to access data.\n",
+    "  - [pooch](https://www.fatiando.org/pooch/latest/) is a good candidate when sample files are needed.\n",
+    "\n",
+    "### Code formatting\n",
+    "\n",
+    "- Line lengths should be limited to avoid horizontal scroll bars when viewing documentation pages in a browser or in Jupyter.\n",
```

Comment:
In addition to limiting code line length, I think we should also add somewhere that in the markdown sections, the convention of one sentence per line should be used.

Reply:
Is this the right place? This document is about readability for users / doc pages.

Reply:
This is in the developer docs. If you are a new developer, and you start writing notebooks, you will get this as a comment when someone reviews your notebooks. I think it's an ok place to have it?

Reply:
👍 I don't mind adding it.